### PR TITLE
memory.events handling rework

### PR DESCRIPTION
**Pull request by Zyqsempai**
Small rework of the memory.events handler in order to implement OOM handler in main containerD repo.

During the memory.events handler implementations, were made few mistakes, this PR fixes them.

Signed-off-by: Boris Popovschi <zyqsempai@mail.ru>

### CODE REVIEW

**Comment by Zyqsempai** [comment]
@AkihiroSuda @crosbymichael PTAL

File: v2/manager.go [hunk 1]
```diff
@@ -526,20 +526,19 @@ func (c *Manager) freeze(path string, state State) error {
 	}
 }
 
-func (c *Manager) MemoryEventFD() (uintptr, error) {
+func (c *Manager) MemoryEventFD() (int, uint32, error) {
```
Comment (AkihiroSuda) [inline v2/manager.go]: needs godoc

**Comment by Zyqsempai** [comment]
@AkihiroSuda Added goDoc and updated PR description, related to "how to test it" i tried to add test but I wasn't able to properly simulate OOM behavior for process in the test.
Maybe you have some suggestions, how to test it properly, coz I am more than agree that we need few tests here.

**Comment by Zyqsempai** [comment]
@AkihiroSuda Can we move forward this one, coz implementation of the OOM handler in containerD main repo depends on this one.

**Comment by AkihiroSuda** [comment]
@fuweid PTAL?

File: v2/manager.go (hunk 1 above)
Comment (AkihiroSuda) [reply v2/manager.go]: > (int, uint32

why different types?

Comment (Zyqsempai) [reply v2/manager.go]: Coz this is the base types for file descriptor and watch descriptor, at least `syscall.InotifyRmWatch()` expects it in that way.

**Pull request merged**